pkg/operator/cassandra/nodetool/nodetool.go

```diff
@@ -0,0 +1,184 @@
+package nodetool
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+
+	"github.com/pborman/uuid"
+	"k8s.io/apimachinery/pkg/util/sets"
+
+	"github.com/rook/rook/pkg/operator/cassandra/nodetool/client"
+	"github.com/rook/rook/pkg/operator/cassandra/nodetool/version"
+)
+
+// NodeState represents the cluster membership state of a C* node.
+type NodeState string
+
+const (
+	NodeStateUnknown NodeState = "Unknown"
+	NodeStateNormal  NodeState = "Normal"
+	NodeStateLeaving NodeState = "Leaving"
+	NodeStateJoining NodeState = "Joining"
+	NodeStateMoving  NodeState = "Moving"
+)
+
+// NodeState represents the reachability of a C* node from the
+// perspective of the node answering the query.
+type NodeStatus string
+
+const (
+	NodeStatusUnknown NodeStatus = "Unknown"
+	NodeStatusUp      NodeStatus = "Up"
+	NodeStatusDown    NodeStatus = "Down"
+)
+
+type Node struct {
+	Host   string
+	ID     uuid.UUID
+	State  NodeState
+	Status NodeStatus
+	Local  bool
+}
+
+type NodeMap map[string]*Node
+
+func (nm NodeMap) LocalNode() *Node {
+	for _, node := range nm {
+		if node.Local {
+			return node
+		}
+	}
+	return nil
+}
+
+type Interface interface {
+	Status() (NodeMap, error)
+	Version() (*version.Version, error)
+}
+
+type tool struct {
+	client client.Interface
+}
+
+var _ Interface = &tool{}
+
+func New(client client.Interface) Interface {
+	return &tool{
+		client: client,
+	}
+}
+
+func NewFromURL(u *url.URL) Interface {
+	return New(client.New(u, &http.Client{}))
+}
+
+func setsIntersect(setsToCheck ...sets.String) bool {
+	all := sets.NewString()
+	totalLength := 0
+	for _, s := range setsToCheck {
+		totalLength += s.Len()
+		all = all.Union(s)
+	}
+	return all.Len() != totalLength
+}
+
+// Status generates a summary of the status of every C* node in the cluster.
+// From the perspective of the local node.
+//
+// It is intended to produce identical information to the `nodetool status` utility.
```

Comment:
the ceph operator shells out (exec) to the ceph binaries instead of duplicating their functionality in golang code.  Should we be considering that here, should we shell out to `nodetool` instead of writing all this golang code?